### ISSUE TITLE
This does not work with Gulp

### ISSUE THREAD

**Issue by ehmicky** [sindresorhus/get-stream#103]
`get-stream` does not currently work with streams created by `gulp.src()`, but that's a bug with Gulp, not `get-stream`. I am creating this issue for other people encountering the same problem.

`gulp.src()` uses [`vinyl-fs`](https://github.com/gulpjs/vinyl-fs/blob/c08ab4b25f1aa003d167464e9f63aa42b69c8fe9/lib/src/index.js#L43), which uses [`to-through`](https://github.com/gulpjs/to-through/blob/54c21fd08cf9f0d0c0bcd9d536d3715ec2719500/index.js#L3), which uses [`streamx`](https://github.com/mafintosh/streamx). Gulp currently uses `vinyl-fs@v3`, which uses an old version of `streamx` that lacks `Readable[Symbol.asyncIterator]`. `vinyl-fs@v4` fixes this, but the latest Gulp (`4.0.2`) does not use that version.

In a nutshell, Gulp must upgrade `vinyl-fs`. They have not released in 4 years, but it seems like there has been some moderate activity latetely.
